api/R/param_query.R

```diff
@@ -0,0 +1,38 @@
+#' Run prepared query
+#'
+#' @param con Database connection, as created by [RPostgres::dbConnect]
+#' @param query Query template (character, length 1)
+#' @param params Query parameters (unnamed list)
+#' @return Query result, as `data.frame`
+#' @author Alexey Shiklomanov
+#' @export
+param_query <- function(con, query, params) {
```

Comment:
I agree that renaming this function and documenting it better are good ideas. I'll do that now.

> having an API function that allows arbitrary queries to be run makes me very nervous

I can see where you're coming from, but I think these sorts of security concerns are better handled by constraints to BETY itself. I designed this API as much (maybe even more) for PEcAn developers as for new users, and wanted to leave the option open for more advanced database management. My view is that users that have the permissions to perform any database operations, should be able to perform those operations. My vision was that, in setting up a truly user-facing PEcAn Docker stack, we secure BETY with a stronger username/password combination than `bety/bety`, and then provide a `guest`, `api`, or other user that has broad `SELECT` permissions but very limited `INSERT/DELETE/UPDATE` permissions (e.g. only `INSERT` into the `workflows` table and maybe `inputs` and `dbfiles`). Then, we wouldn't have to obsess about security in our database code, since most of those operations would be forbidden anyway.

Reply:
Better documentation and clearer names added in b752a264a.

Reply:
I agree with all of this, but there's the additional security concern that relying on a direct DB connection also means opening up a public port on the database server (which the BETY web api doesn't require) and causes an authentication/permissions disconnect between the BETY user settings and the postgres DB settings. i.e. with the BETY web-based API we can control what access level a user has, but those permissions are not mirrored in their PostGRES login/password permissions

I agree that switching this over is unrealistic for this PR, but I think the better solution is to expand the capabilities of the BETY interface

Reply:
Point taken. I definitely agree that moving towards a 100% BETY API would also have the advantage of eliminating the dependency on `RPostgres` and all the query code in here. I would be happy to contribute to BETY API development, but definitely not till after AGU.